### ISSUE TITLE
convert_Wavelength2Energy() fails on a RLum.Data.Spectrum object if order=TRUE

### ISSUE THREAD

**Issue by mcol** [R-Lum/Luminescence#133]
Running this snippet:
```R
lambda <- seq(400,800,50)
data <- matrix(data = rep(1, 2 * length(lambda)), ncol = 2)
rownames(data) <- lambda
colnames(data) <- 1:ncol(data)
object <- set_RLum(class = "RLum.Data.Spectrum", data = data[,1,drop = FALSE])
convert_Wavelength2Energy(object, order = TRUE)
```
fails with:
```console
Error in (function (cl, name, valueClass)  : 
  assignment of an object of class “numeric” is not valid for @‘data’ in an object of class “RLum.Data.Spectrum”; is(value, "matrix") is not TRUE
```

**Comment by mcol** [comment]
The error occurs at this line:
```R
object@data <- object@data[order(as.numeric(rownames(object@data))), ]
```
because if `object` is of the `RLum.Data.Spectrum` class, it is a 1-column matrix, and the line above implicitly turns it into a vector, so the assignment fails. Adding `drop=FALSE` fixes the issue.

**Comment by RLumSK** [comment]
Fixes with PR #134
